### ISSUE TITLE
Create Action to Close Orders [MASF]

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#2443]
_Imported from trac ticket 2443.
Created by **aadams**
Opened in trac: Aug 12, 2013, 1:30:06 PM
Last modified in trac: Mar 23, 2016, 11:50:21 AM_
***
_by aadams:_
Is there a way to set a process to run daily to auto-finalize MCR’s and MDB’s? If not, can we at least set up an Action to be used for this? We have not created actions on our own, but I think we have documentation on how to key?
___
Create an action to close all order types.  Start with MCR/MDB at Miskelly's request.

MASF screen.

**Comment by kentbower** [comment]
_**sjohnson** trac comment on Oct 15, 2013, 4:10:09 PM:_

Does this action need to do any validation on the queried orders?  On train, I was able to close a MCR with a negative balance leading me to believe there isn't any, however I did see these 2 comments in the MCR.

```
superuser changed percent of sale 2 from [0] to [None]
superuser closed order 0928304NSRY
```

**Comment by kentbower** [comment]
_**sjohnson** trac comment on Oct 16, 2013, 12:00:20 PM:_

**Implementation Considerations**
 - How do you communicate errors?  (How do you communicate warnings?)  Our current list in a huge string showing all the errors that occurred would not be a good design.  
 - How do you manage database transactions?  (We should not be doing commits in server code ever, if possible, but if an order has an error, how will that be dealt with?  If we need to start doing commits for selection of multiple records for an action, that should be a framework change so there aren't commits in the general code.)  
 - How do you mimic a GET properly so you can change the status flag to 'F'?  Will this correctly deal with automatic adjustments?  
 - What about Returns, which need user input?

Is there an urgent need now when GERS MASF can still be used?

**Comment by kentbower** [comment]
_**rneville** trac comment on Mar 31, 2014, 9:51:27 AM:_

Ticket #237 has been marked as a duplicate of this ticket.

**Comment by kentbower** [comment]
_**kbower** trac comment on Mar 2, 2016, 10:38:02 AM:_

(Changeset [[7202]](../commit/d98d1328f330e7e931b8a38df0734acd8d4ff938))
* pull all the (way too much) common code from RDataGrid.as and RAdvancedDataGrid.as and put in include new file "RDataGridImpl.as" in prep re #2443 framework changes

**Comment by kentbower** [comment]
_**kbower** trac comment on Mar 18, 2016, 8:15:18 AM:_

See prerequisite ticket [#3543](3543)

**Comment by kentbower** [comment]
_**kbower** trac comment on Mar 23, 2016, 11:50:21 AM:_

(Changeset [[7261]](../commit/cd48e27aacb8f8686e047a9503bb437d46de72c4))
* closes #2443, Action to close (possibly multiple) orders #all

**Comment by kentbower** [comment]
_**kbower** trac status change on Mar 23, 2016, 11:50:21 AM:_ **closed**
